Test for 1-my_list.py


case 0: import
>>> MyList = __import__('1-my_list').MyList
>>>


case 1: common case

>>> my_list = MyList()
>>> my_list.append(7)
>>> my_list.append(4)
>>> my_list.append(1)
>>> my_list.append(3)
>>> my_list.append(5)
>>> print(my_list)
[7, 4, 1, 3, 5]
>>> my_list.print_sorted()
[1, 3, 4, 5, 7]
>>> print(my_list)
[7, 4, 1, 3, 5]
>>>


case 2: not defined object

>>> MyList = __import__('1-my_list').MyList
>>> my_list2.print_sorted()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'my_list2' is not defined
>>>


case 3: passed a list

>>> a = [2, 6, 4]
>>> my_list = MyList(a)
>>> my_list.print_sorted()
[2, 4, 6]
>>>


case 4: empty list

>>> a = []
>>> my_list = MyList(a)
>>> my_list.print_sorted()
[]
>>>


case 5: unique number

>>> my_list = MyList()
>>> my_list.append(1)
>>> my_list.print_sorted()
[1]
>>> print(my_list)
[1]
>>>


case 6: same number

>>> my_list2 = MyList()
>>> my_list2.append(2)
>>> my_list2.append(2)
>>> my_list2.append(2)
>>> my_list2.append(2)
>>> my_list2.print_sorted()
[2, 2, 2, 2]
>>>


case 7: negative numbers

>>> MyList = __import__('1-my_list').MyList
>>> my_list = MyList()
>>> my_list.append(4)
>>> my_list.append(2)
>>> my_list.append(42)
>>> my_list.append(-1)
>>> my_list.append(-6)
>>> my_list.print_sorted()
[-6, -1, 2, 4, 42]
>>>
